src/libasr/codegen/asr_to_c.cpp

```diff
@@ -779,6 +790,12 @@ R"(
     }
 
     void visit_Module(const ASR::Module_t &x) {
+        if (startswith(x.m_name, "lfortran_intrinsic_")) {
+            intrinsic_module = true;
+        } else {
+            intrinsic_module = false;
+        }
```

Comment:
```suggestion
        intrinsic_module = x.m_intrinsic;
```
I think here we can also utilize the `x.m_intrinsic` data member of `Module_t`.